### PR TITLE
feat: update goreleaser develop build

### PR DESCRIPTION
**Pull request by ajgrande924**
- bump goreleaser to 1.15.x
- cgo disabled but allow for potential native deps in the future
- fix goreleaser deprecation warnings

### CODE REVIEW

File: .goreleaser.develop.yaml [hunk 1]
```diff
@@ -64,8 +64,8 @@ dockers:
     use: buildx
     goos: linux
     goarch: amd64
-    extra_files:
-      - tmp/linux_amd64/libs
+    # extra_files:
```
Comment (chainchad) [inline .goreleaser.develop.yaml:67]: Can we add a comment on why we would want to uncomment these in the future?

File: .goreleaser.develop.yaml [hunk 2]
```diff
@@ -174,11 +174,12 @@ docker_signs:
   - artifacts: all
     stdin: "{{ .Env.COSIGN_PASSWORD }}"
 
-archives:
-  - files:
-      - src: tmp/{{ .Os }}_{{ .Arch }}/libs/*
-        dst: libs
-        strip_parent: true
+archives: []
+  # - rlcp: true
```
Comment (chainchad) [inline .goreleaser.develop.yaml:178]: Needs a comment here too to explain why this is commented and whey/when we would uncomment if we want to leave it

**Review by chainchad — DISMISSED** [review]
Looks good, but would add some comments (about the comments) 👍

File: .goreleaser.develop.yaml (hunk 1 above)
Comment (ajgrande924) [reply .goreleaser.develop.yaml:68]: added

File: .goreleaser.develop.yaml (hunk 2 above)
Comment (ajgrande924) [reply .goreleaser.develop.yaml:183]: added

**Review by chainchad — APPROVED** [review]

**Review by jmank88 — APPROVED** [review]

**Pull request merged**